packages/browser/src/index.test.ts

```diff
@@ -1,17 +1,50 @@
-import { createRequester } from '@logto/js';
+import { createRequester, generateCodeChallenge, parseUriParameters, QueryKey } from '@logto/js';
 
-import LogtoClient from './index';
+import LogtoClient, { getLogtoKey, LogtoSignInSessionItem } from './index';
+
+const clientId = 'client_id_value';
+const endpoint = 'https://logto.dev';
+const authorizationEndpoint = `${endpoint}/oidc/auth`;
+const requester = createRequester();
+
+jest.mock('@logto/js', () => ({
+  ...jest.requireActual('@logto/js'),
+  fetchOidConfig: async () => ({
+    authorization_endpoint: authorizationEndpoint,
+  }),
+}));
 
 describe('LogtoClient', () => {
   test('constructor', () => {
-    expect(
-      () =>
-        new LogtoClient({
-          endpoint: 'https://logto.dev',
-          clientId: 'client_id_value',
-          usingPersistStorage: false,
-          requester: createRequester(),
-        })
-    ).not.toThrow();
+    expect(() => new LogtoClient({ endpoint, clientId, requester })).not.toThrow();
+  });
+
+  describe('signInWithBrowser', () => {
+    test('sessionStorage and signInUri should contain correct properties', async () => {
+      const logtoClient = new LogtoClient({ endpoint, clientId, requester });
+      const originalRedirectUri = 'http://localhost:3000/callback';
+      await logtoClient.signInWithBrowser(originalRedirectUri);
+
+      const item = sessionStorage.getItem(getLogtoKey(clientId)) ?? '{}';
+      const logtoSignInSessionItem = JSON.parse(item) as LogtoSignInSessionItem;
+
+      const state = logtoSignInSessionItem?.state;
+      const codeVerifier = logtoSignInSessionItem?.codeVerifier;
+      const codeChallenge = await generateCodeChallenge(codeVerifier);
+      const redirectUri = logtoSignInSessionItem?.redirectUri;
```

Comment:
And based on your current code, the `logtoSignInSessionItem` is forced to be an instance of `LogtoSignInSessionItem`, and it will never be optional. Thus the `?` makes no sense in these sentences.
In fact, we could improve our linter to auto-fix these problems

Reply:
Improved. @demonzoo  PTAL.